samcli/commands/local/lib/local_api_service.py

```diff
@@ -72,7 +72,8 @@ def start(self):
         LOG.info(
             "You can now browse to the above endpoints to invoke your functions. "
             "You do not need to restart/reload SAM CLI while working on your functions, "
-            "changes will be reflected instantly/automatically. You only need to restart "
+            "changes will be reflected instantly/automatically (note: you will still need to "
```

Comment:
I think there's actually a little more to it than that. This message was correct for certain runtimes before `sam build`. For example, if you're using Node.js or Python without running `sam build` then this message is true. As we evolved it seems like some of our messaging hasn't.

Maybe something along the lines of?
```suggestion
            changes will be reflected instantly/automatically. If you used sam build before running local commands, you will need to re-run sam build for the changes to be picked up. You only need to restart
```

Reply:
@mildaniel this makes sense!